src/core_plugins/kibana/public/discover/controllers/get_painless_error.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get } from 'lodash';
+
+export function getPainlessError(error: Error) {
+  const rootCause: Array<{ lang: string; script: string }> | undefined = get(
+    error,
+    'resp.error.root_cause'
+  );
+
+  if (!rootCause) {
+    return;
+  }
+
+  const { lang, script } = rootCause[0];
```

Comment:
__optional nit:__ since you're already using destructuring you can use a bit more of it :)
```ts
const [{ lang, script }] = rootCause;
```